cmd/kube-proxy/app/server.go

```diff
@@ -149,6 +149,12 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&o.master, "master", o.master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
 	fs.StringVar(&o.hostnameOverride, "hostname-override", o.hostnameOverride, "If non-empty, will use this string as identification instead of the actual hostname.")
 	fs.StringVar(&o.config.IPVS.Scheduler, "ipvs-scheduler", o.config.IPVS.Scheduler, "The ipvs scheduler type when proxy mode is ipvs")
+	fs.StringVar(&o.config.ShowHiddenMetricsForVersion, "show-hidden-metrics-for-version", o.config.ShowHiddenMetricsForVersion,
+		"The previous version for which you want to show hidden metrics. "+
+			"Only the previous minor version is meaningful, other values will not be allowed. "+
```

Comment:
I know this is part of a larger effort, but this comment is confusing to me.

If it only allows the previous version, why do I have to specify a version?

Reply:
First of all, this flag is an escape hatch, it intends to be used by admins if they were not able to react to the earlier deprecation warnings.

Second of all, we don't want admins to be excessive depend on this flag, otherwise, they may fail to notice deprecate message from the next release.

Here is an example: [why-not-bool-flag](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/20190404-kubernetes-control-plane-metrics-stability.md#why-not-bool-flag)

@thockin